### ISSUE TITLE
[drogon] build failure

### ISSUE THREAD

**Issue by dustinnieffenegger** [microsoft/vcpkg#12809]
**Host Environment**
- OS: Windows 8.1 (32-bit)
- Compiler: 2020.06.15-nohash

**To Reproduce**
Typed `vcpkg install drogon` with freshly installed vcpkg and waited for dependencies to build and install -- all we successful until drogon. It seems to not be linking zlib.

**Failure message to standard output** 
```
C:\dev\vcpkg>vcpkg install drogon
Computing installation plan...
The following packages will be built and installed:
    drogon[core]:x86-windows
Detecting compiler hash for triplet x86-windows...
Starting package 1/1: drogon:x86-windows
Building package drogon[core]:x86-windows...
Could not locate cached archive: C:\Users\Home\AppData\Local\vcpkg\archives\20\2
0467c46fb1afea51d6bf0fe3c27c05b54b20b54.zip
-- Using cached C:/dev/vcpkg/downloads/an-tao-drogon-v1.0.0-beta18.tar.gz
-- Cleaning sources at C:/dev/vcpkg/buildtrees/drogon/src/0.0-beta18-aa4dfb0262.
clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source C:/dev/vcpkg/downloads/an-tao-drogon-v1.0.0-beta18.tar.gz
-- Applying patch vcpkg.patch
-- Applying patch drogon_ctl_install.patch
-- Using source at C:/dev/vcpkg/buildtrees/drogon/src/0.0-beta18-aa4dfb0262.clea
n
-- Configuring x86-windows-dbg
-- Configuring x86-windows-rel
-- Building x86-windows-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:139 (message):
    Command failed: C:/dev/vcpkg/downloads/tools/cmake-3.17.2-windows/cmake-3.17
.2-win32-x86/bin/cmake.exe --build . --config Debug --target install -- /p:VCPkg
LocalAppDataDisabled=true /p:UseIntelMKL=No /m
    Working Directory: C:/dev/vcpkg/buildtrees/drogon/x86-windows-dbg
    See logs for more information:
      C:\dev\vcpkg\buildtrees\drogon\install-x86-windows-dbg-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_build_cmake.cmake:91 (vcpkg_execute_build_process)
  scripts/cmake/vcpkg_install_cmake.cmake:24 (vcpkg_build_cmake)
  ports/drogon/portfile.cmake:19 (vcpkg_install_cmake)
  scripts/ports.cmake:79 (include)


Error: Building package drogon:x86-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: drogon:x86-windows
  Vcpkg version: 2020.06.15-nohash

Additionally, attach any relevant sections from the log files above.
```

**Relevant log files**
[install-x86-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/5043611/install-x86-windows-dbg-out.log)

**Comment by JonLiu1993** [comment]
Hi @dustinnieffenegger ,
I successfully installed drogon on windows10 64bit, for your question, My guess is that you're not linking zlib.lib to your project, or zlib is not properly compiled with drogon

**Comment by ciuczek-dev** [comment]
Build breaks when linking against mariadbclient.lib
Windows 10 64bit
Visual Studio 2019 

[install-x86-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/5423319/install-x86-windows-dbg-out.log)
[](url)

**Comment by NancyLi1013** [comment]
@ciuczek-dev 
Thanks for posting this issue.

The problem caused by the linkage of `openssl` has been fixed via #14196.

**Comment by JonLiu1993** [comment]
@dustinnieffenegger ,This issue has been fixed, you can update vcpkg and try again, if you still have problems, you can reopen this issue
